### PR TITLE
Adding Docker compose and Makefile support for the project

### PR DESCRIPTION
**Pull request by Yash-Handa**
Note, to use Docker containers the following `env` variables are to be modified:

`DB_HOST=db`
`REDIS_ADDRESS=redis:6379`
`DATABASE_URL=postgres://go_template_role:go_template_role456@db:5432/go_template?sslmode\=disable`

because both, `PostgreSQL` and `Redis` are in the same subnet as the `api` and their `IP` is provided by the docker itself.

Another change:
Subscriptions are now working :100:

### CODE REVIEW

**Comment by Yash-Handa** [comment]
Don't merge this PR if https://github.com/wednesday-solutions/go-template/pull/40 has been merge, it breaks this PR. Once https://github.com/wednesday-solutions/go-template/pull/40 is merged I will update this PR to be compatible again and notify in here.

**Comment by Yash-Handa** [comment]
This PR is compatible again, to be precise the Dockerfiles are re-configured to run with `server`  binary rather than `api` binary : )

**Review by praveenchand-wed — APPROVED** [review]

**Comment by praveenchand-wed** [comment]
Hey @Yash-Handa there is one conflict here.

**Pull request merged**
